silkrpc/context_pool.cpp

```diff
@@ -50,7 +53,9 @@ ContextPool::ContextPool(std::size_t pool_size, ChannelFactory create_channel) :
         auto database = std::make_unique<ethdb::kv::RemoteDatabase<>>(*io_context, grpc_channel, grpc_queue.get()); // TODO(canepat): move elsewhere
         auto backend = std::make_unique<ethbackend::BackEnd>(*io_context, grpc_channel, grpc_queue.get()); // TODO(canepat): move elsewhere
         auto tx_pool = std::make_unique<txpool::TransactionPool>(*io_context, grpc_channel, grpc_queue.get()); // TODO(canepat): move elsewhere
-        contexts_.push_back({io_context, std::move(grpc_queue), std::move(grpc_runner), std::move(database), std::move(backend), std::move(tx_pool)});
+        //auto block_cache = std::make_unique<silkrpc::BlockCache>(1024, false);
+        contexts_.push_back({io_context, std::move(grpc_queue), std::move(grpc_runner), std::move(database), std::move(backend), std::move(tx_pool), block_cache});
+        //contexts_.push_back({io_context, std::move(grpc_queue), std::move(grpc_runner), std::move(database), std::move(backend), std::move(tx_pool), std::move(block_cache)});
         SILKRPC_DEBUG << "ContextPool::ContextPool context[" << i << "] " << contexts_[i] << "\n";
```

Comment:
As discussed, we can use the shared cache solution until measurements prove we need a different setup, so you can remove these commented lines